src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs

```diff
@@ -0,0 +1,202 @@
+    /// </summary>
+    public class CachedSchemaRegistryClient : ISchemaRegistryClient, IDisposable
+    {
+        private IRestService restService;
+        private readonly int identityMapCapacity;
+        private readonly Dictionary<int, string> schemaById = new Dictionary<int, string>();
+        private readonly Dictionary<string /*subject*/, Dictionary<string, int>> idBySchemaBySubject = new Dictionary<string, Dictionary<string, int>>();
+        private readonly Dictionary<string /*subject*/, Dictionary<int, string>> schemaByVersionBySubject = new Dictionary<string, Dictionary<int, string>>();
+        
+        /// <summary>
+        ///     The default timeout value for Schema Registry REST API calls.
+        /// </summary>
+        public const int DefaultTimeout = 10000;
+
+        /// <summary>
+        ///     The default maximum capacity of the local schema cache.
+        /// </summary>
+        public const int DefaultMaxCapacity = 1024;
+
+        /// <summary>
+        ///     Initialize a new instance of the SchemaRegistryClient class.
+        /// </summary>
+        /// <param name="config">
+        ///     Configuration properties.
+        /// </param>
+        public CachedSchemaRegistryClient(IEnumerable<KeyValuePair<string, object>> config)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException("config properties must be specified.");
+            }
+
+            var schemaRegistryUrisMaybe = config.Where(prop => prop.Key.ToLower() == "schema.registry.url").FirstOrDefault();
+            if (schemaRegistryUrisMaybe.Value == null)
+            {
+                throw new ArgumentException("schema.registry.url configuration property must be specified.");
+            }
+            var schemaRegistryUris = (string)schemaRegistryUrisMaybe.Value;
+
+            var timeoutMsMaybe = config.Where(prop => prop.Key.ToLower() == "schema.registry.timeout.ms").FirstOrDefault();
+            var timeoutMs = timeoutMsMaybe.Value == null ? DefaultTimeout : (int)timeoutMsMaybe.Value;
+
+            var identityMapCapacityMaybe = config.Where(prop => prop.Key.ToLower() == "schema.registry.cache.capacity").FirstOrDefault();
+            this.identityMapCapacity = identityMapCapacityMaybe.Value == null ? DefaultMaxCapacity : (int)identityMapCapacityMaybe.Value;
+
+            this.restService = new RestService(schemaRegistryUris, timeoutMs);
+        }
+
+        /// <remarks>
+        ///     Ensure memory doesn't explode in the case of incorrect usage.
+        /// </remarks>
+        private void CheckIfCacheFull()
+        {
+            if (
+                this.schemaById.Count + 
+                this.schemaByVersionBySubject.Sum(x => x.Value.Count) +
+                this.idBySchemaBySubject.Sum(x => x.Value.Count)
+                    >= identityMapCapacity)
+            {
+                throw new OutOfMemoryException("Local schema cache maximum capacity exceeded");
```

Comment:
I put in a todo. I think if the cache fills up this will most likely be unexpected, in which case failing fast is probably actually preferable to evicting. anyway, it's more work, and the java client just throws an exception, so this is good enough for an mvp.